index.bs

```diff
@@ -1736,6 +2079,230 @@ The <dfn method for=StorageManager>getDirectory()</dfn> method steps are:
 
 </div>
 
+# Observing file system change events # {#observing-the-file-system}
+
+## The {{FileSystemObserver}} interface ## {#api-filesystemobserver}
+
+<xmp class=idl>
+dictionary FileSystemObserverObserveOptions {
+  boolean recursive = false;
+};
+
+callback FileSystemObserverCallback = undefined (
+  sequence<FileSystemChangeRecord> records,
+  FileSystemObserver observer
+);
+
+[
+    Exposed=(DedicatedWorker,SharedWorker,Window),
```

Comment:
Why not `(Worker, Window)`, `FileSystemHandle` is probably exposed in service worker contexts too?

Reply:
A separate API would be needed for service workers. The service worker becomes inactive after 30 seconds of no events. The change events received in the `FileSystemObserver` callback would not count as a service worker event. There are also privacy considerations around allowing a site to observe file changes without the user being aware.